src/troute-network/troute/AbstractRouting.py

```diff
@@ -0,0 +1,323 @@
+from abc import ABC, abstractmethod
+import logging
+import yaml
+import json
+import xarray as xr
```

Comment:
double check that xarray is properly documented as a dependency in setup.py/cfg and requirements.txt

I think pyyaml is another that may be missing from at least requirements.txt